src/main/java/com/_up/megastore/config/SecurityConfig.java

```diff
@@ -40,7 +42,8 @@ public SecurityFilterChain securityFilterChain(HttpSecurity httpSecurity) throws
             .requestMatchers(HttpMethod.OPTIONS).permitAll()
             .requestMatchers(ALLOWED_TO_ADMINISTRATORS_URLS).hasRole(Role.ADMIN.name()))
         .sessionManagement(sessionManager -> sessionManager.sessionCreationPolicy(SessionCreationPolicy.STATELESS))
-        .httpBasic(Customizer.withDefaults())
+        .httpBasic(httpSecurityHttpBasicConfigurer ->
+                httpSecurityHttpBasicConfigurer.authenticationEntryPoint(authenticationEntryPoint))
         .authenticationProvider(authenticationProvider)
         .addFilterAfter(jwtAuthenticationFilter, BasicAuthenticationFilter.class);
```

Comment:
Siguiendo el comentario anterior, esto deberiamos modificarlo asi

```suggestion
        .addFilterAfter(jwtAuthenticationFilter, BasicAuthenticationFilter.class)
            .exceptionHandling().authenticationEntryPoint(jwtAuthorizationEntryPoint());
```